server/etcdserver/api/membership/cluster.go

```diff
@@ -550,7 +558,7 @@ func (c *RaftCluster) PromoteMember(id types.ID, shouldApplyV3 ShouldApplyV3) {
 			// promotion request had been applied to v3store, but not saved
 			// to v2snapshot yet when in 3.5. Once upgrading to 3.6, the
 			// patch here ensure the issue can be automatically fixed.
-			if m.IsLearner {
+			if m != nil && m.IsLearner {
```

Comment:
Technically message `"ignore already promoted member"` should be different.

Reply:
Updated & thx.